src/Stream.php

```diff
@@ -676,6 +712,61 @@ public function toArray(): array
         return $result;
     }
 
+    /**
+     * Writes iterable source to the file line by line.
+     *
+     * Elements of the iterable source must be stringifiable.
+     *
+     * @param resource $file
+     * @param string $separator (optional) inserted between each item. Ex: ', ' for 1, 2, 3, ...
+     * @param string $prefix    (optional) prepended to string
+     * @param string $suffix    (optional) appended to string
+     *
+     * @return void
+     */
+    public function toFile($file, string $separator = "\n", string $prefix = '', string $suffix = ''): void
+    {
+        ResourceHelper::checkIsValid($file);
+
+        $firstIteration = true;
+
+        fputs($file, $prefix);
+
+        foreach ($this->iterable as $line) {
+            if ($firstIteration) {
+                $firstIteration = false;
+            } else {
+                $line = $separator . \strval($line);
+            }
+
+            fputs($file, \strval($line));
+        }
+
+        fputs($file, $suffix);
+    }
+
+    /**
+     * Writes iterable source to the file line by line.
+     *
+     * Elements of the iterable source must be array of scalars.
+     *
+     * @param resource $file
+     * @param string $separator
+     * @param string $enclosure
+     * @param string $escape
+     *
+     * @return void
+     */
+    public function toCsvFile($file, string $separator = ',', string $enclosure = '"', string $escape = '\\'): void
```

Comment:
If `toFile` has an optional _header_ text, then I think it would be reasonable to expect `toCsvFile` to also have it. I think it is a common case where you will have a data structure of all the rows of the CSV but that data structure will not contain the header row. And to make a CSV make sense in Excel or Google Sheets, you usually are going to want a header row defining the columns. Since the separator etc. is passed in, then it probably makes sense for the optional `headers` to also be an array. Thoughts?

Reply:
Done.